test/Sentry.Serilog.Tests/SentrySerilogConfigurationSettingsIntegrationTests.cs

```diff
@@ -0,0 +1,209 @@
+                                ""debug"": false,
+                                ""diagnosticsLevel"": ""Debug"",
+                                ""reportAssemblies"": false,
+                                ""deduplicateMode"": ""All"",
+                                ""initializeSdk"": true,
+                                ""minimumBreadcrumbLevel"": ""Verbose"",
+                                ""minimumEventLevel"": ""Error""
+                            }
+                        }
+                    ]
+                }
+            }")]
+        [InlineData(
+            "No properties are provided",
+            @"{
+                ""Serilog"": {
+                    ""Using"": [
+                        ""Serilog"",
+                        ""Sentry""
+                    ],
+                    ""WriteTo"": [{
+                            ""Name"": ""Sentry""
+                        }
+                    ]
+                }
+            }")]
+        [InlineData(
+            "Some properties are provided",
+            @"{
+                ""Serilog"": {
+                    ""Using"": [
+                        ""Serilog"",
+                        ""Sentry""
+                    ],
+                    ""WriteTo"": [{
+                            ""Name"": ""Sentry"",
+                            ""Args"": {
+                                ""sendDefaultPii"": false,
+                                ""isEnvironmentUser"": false
+                            }
+                        }
+                    ]
+                }
+            }")]
+        [InlineData(
+            "Properties are provided out of order",
+            @"{
+                ""Serilog"": {
+                    ""Using"": [
+                        ""Serilog"",
+                        ""Sentry""
+                    ],
+                    ""WriteTo"": [{
+                            ""Name"": ""Sentry"",
+                            ""Args"": {
+                                ""release"": ""0.0.1-pre"",
+                                ""minimumBreadcrumbLevel"": ""Verbose"",
+                                ""reportAssemblies"": false,
+                                ""isEnvironmentUser"": false,
+                                ""attachStackTrace"": false,
+                                ""shutdownTimeout"": ""00:00:09"",
+                                ""environment"": ""some environment""
+                            }
+                        }
+                    ]
+                }
+            }")]
+        public void SentrySerilogOptionsIsConfiguredFromJsonProperly(string when, string json)
+        {
+            Assert.False(string.IsNullOrWhiteSpace(when));
+
+            // This is used as the resulting object / output of the method being tested
+            var actualSentrySerilogOptions = new SentrySerilogOptions();
+
+            var configurationSections = GetConfigurationSectionsFromJson(json);
```

Comment:
Would it be possible to avoid the reflection approach and instead simply "init" the SDK with the Serilog integration and then get a hold of the `SerilogSentryOptions` after the libraries binding values and asserting those?

It's quite a bit of supporting code and reflection (brittle) so I'm not sure it's really worth it.

If this is too hard, perhaps then just programatically calling your new method passing all parameters and making sure they all bind to the SentryOptions property you'd expect. And a second call passing null and verifying nothing was defined. What do you think?

Reply:
I agree with the brittle nature of this. I struggled to find the best way to approach this--the only way to get at the underlying SentrySerilogOptions object is to perform reflection on the Log.Logger, and then dig through the 'RestrictedSinks' and try to convert said object back into something usable. This is even _more_ brittle than what I had as it would rely more on the Serilog project never changing.

I opted to do as you suggested and went the manually-call-the-method route. I performed a 'zero, one, or many' test including putting the parameters out of order, and also did an 'all parameters' type of test. 

I used a hack to do a deep equals--is there a library that should be used in its place? I couldn't find any similar item and the problem with Assert.Equals in XUnit is that it doesn't deeply compare objects--it seems to only compare references? Or perhaps I'm missing something?